### PR TITLE
Diagnosis tree norsk

### PR DESCRIPTION
**Pull request by joddern**
The diagnosis.ts router has two endpoints: "createFileAndEmbedd" which can use the "dsm_norsk_nettside.html" file from https://finnkode.ehelse.no/#icd10/1/0/0/-1, to parse it into categories and diagnoses and then embeds this into a weaviate vector store. Also has functions for writing to json file.

The other endpoint: "queryTheDatabase" is supposed to be used in the chat.tsx file as a "second module" where symptoms can be plotted in and followup questions must be answered to pinpoint a diagnoses suggestion. This is only in norwegian for now (the website above is norwegian).

### CODE REVIEW

**Comment by joddern** [comment]
[dsm_norsk_nettside_textformat.txt](https://github.com/Konfidens/freud/files/12228736/dsm_norsk_nettside_textformat.txt)

File: src/components/Chat.tsx [hunk 1]
```diff
@@ -124,11 +128,51 @@ const Chat = ({ messages, setMessages, categories }: Prop) => {
...
+    }
+  })
+
+  function toggleSymptomsMode() {
+    setDiagnosisMode(!diagnosisMode);
   }
 
   return (
     <>
+      <Button onClick={toggleSymptomsMode}> Toggle symptoms-mode </Button>
```
Comment (fredrikrab) [inline src/components/Chat.tsx:175]: How can be make the UI for this feature neat?

File: src/server/api/routers/diagnosis.ts [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+    await createDiagnosisVectorStoreFromDocuments("DSM", docs, embeddings);
+  }),
+
+  queryTheDatabase: publicProcedure
+
+    .input(z.object({ qa: z.array(z.string()), symptoms: z.array(z.string()) }))
+
+    .mutation(async ({ input }) => {
+      console.debug("\nQUERYING!");
+      const NUM_SOURCES = 5;
```
Comment (fredrikrab) [inline src/server/api/routers/diagnosis.ts:68]: Move "magic constants" to the very top?

File: src/server/api/routers/diagnosis.ts [hunk 3]
```diff
@@ -0,0 +1,467 @@
...
+        // Add the resulting symptom if GPT returned anything.
+        // Length > emptyMargin, because GPT may return some spacings instead of absolutely no characters
+        const emptyMargin = 3;
+
+        if (followUpSymptomToAdd.length > emptyMargin) {
+          newSymptom = followUpSymptomToAdd;
+        } else {
+          newSymptom = "";
+        }
+        console.debug("\nnewSymptom: ", newSymptom);
```
Comment (fredrikrab) [inline src/server/api/routers/diagnosis.ts:113]: Remove debug statements before merging to dev?

**Review by fredrikrab — CHANGES_REQUESTED** [review]
Very cool! I added the three comments we also talked about in person for reference

File: src/components/Chat.tsx [hunk 4]
```diff
@@ -124,11 +128,51 @@ const Chat = ({ messages, setMessages, categories }: Prop) => {
       content: query,
     };
     setMessages([...messages, message]);
-    mutation.mutate({ messages: [...messages, message], categories });
+    if (!diagnosisMode) {
+      mutation.mutate({ messages: [...messages, message], categories });
+    }
+    else {
+      console.log("QUERYING!");
```
Comment (davidlekve) [inline src/components/Chat.tsx:135]: please remove

File: src/components/Chat.tsx (hunk 1 above)
Comment (joddern) [reply src/components/Chat.tsx:175]: I don't know

File: src/components/MessageComponent.tsx [hunk 5]
```diff
@@ -93,10 +95,13 @@ const MessageComponent = ({ message, children }: Prop) => {
             {children}
             {formatLinks(message.content)}
           </div>
-          {message.sources ?
-            <SourceList sources={message.sources} scrollToId={scrollToId} setScrollToId={setScrollToId} />
-            : null
-          }
+          {message.sources ? (
```
Comment (davidlekve) [inline src/components/MessageComponent.tsx:98]: Change to &&

**Review by fredrikrab — APPROVED** [review]

**Pull request merged**